plugins/inputs/proxysql/proxysql.go

```diff
@@ -0,0 +1,354 @@
+	rows, err := db.Query(connectionPool)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	var (
+		srv_host, status, hostgroup, srv_port        string
+		connused, connfree, connok, connerr, queries uint64
+		bytes_data_sent, bytes_data_recv, latency_us uint64
+	)
+
+	servtag := getDSNTag(serv)
+
+	for rows.Next() {
+		err = rows.Scan(
+			&hostgroup,
+			&srv_host,
+			&srv_port,
+			&status,
+			&connused,
+			&connfree,
+			&connok,
+			&connerr,
+			&queries,
+			&bytes_data_sent,
+			&bytes_data_recv,
+			&latency_us,
+		)
+
+		if err != nil {
+			return err
+		}
+
+		tags := map[string]string{
+			"server":    servtag,
+			"hostgroup": hostgroup,
+			"srv_host":  srv_host,
+			"srv_port":  srv_port,
+		}
+
+		fields := make(map[string]interface{})
+		fields["status"] = status
+		fields["connused"] = connused
+		fields["connfree"] = connfree
+		fields["connok"] = connok
+		fields["connerr"] = connerr
+		fields["queries"] = queries
+		fields["bytes_data_sent"] = bytes_data_sent
+		fields["bytes_data_recv"] = bytes_data_recv
+		fields["latency_us"] = latency_us
+
+		acc.AddFields("proxysql_connection_pool", fields, tags)
+	}
+
+	return nil
+}
+
+func (p *Proxysql) gatherCommandsCounters(db *sql.DB, serv string, acc telegraf.Accumulator) error {
+	rows, err := db.Query(commandsCounters)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	var (
+		command                                                   string
+		total_time_us, total_cnt, cnt_100us, cnt_500us, cnt_1ms   uint64
+		cnt_5ms, cnt_10ms, cnt_50ms, cnt_100ms, cnt_500ms, cnt_1s uint64
+		cnt_5s, cnt_10s, cnt_infs                                 uint64
+	)
+
+	servtag := getDSNTag(serv)
+
+	for rows.Next() {
+		err = rows.Scan(
+			&command,
+			&total_time_us,
+			&total_cnt,
+			&cnt_100us,
+			&cnt_500us,
+			&cnt_1ms,
+			&cnt_5ms,
+			&cnt_10ms,
+			&cnt_50ms,
+			&cnt_100ms,
+			&cnt_500ms,
+			&cnt_1s,
+			&cnt_5s,
+			&cnt_10s,
+			&cnt_infs,
+		)
+
+		if err != nil {
+			return err
+		}
+
+		tags := map[string]string{
+			"server":  servtag,
+			"command": command,
+		}
+
+		fields := make(map[string]interface{})
+		fields["total_time_us"] = total_time_us
+		fields["total_cnt"] = total_cnt
+		fields["cnt_100us"] = cnt_100us
+		fields["cnt_500us"] = cnt_500us
+		fields["cnt_1ms"] = cnt_1ms
+		fields["cnt_5ms"] = cnt_5ms
+		fields["cnt_10ms"] = cnt_10ms
+		fields["cnt_50ms"] = cnt_50ms
+		fields["cnt_100ms"] = cnt_100ms
+		fields["cnt_500ms"] = cnt_500ms
+		fields["cnt_1s"] = cnt_1s
+		fields["cnt_5s"] = cnt_5s
+		fields["cnt_10s"] = cnt_10s
+		fields["cnt_infs"] = cnt_infs
+
+		acc.AddFields("proxysql_commands_counters", fields, tags)
+	}
+
+	return nil
+}
+
+func dsnAddTimeout(dsn string) (string, error) {
+	conf, err := mysql.ParseDSN(dsn)
+	if err != nil {
+		return "", err
+	}
+
+	if conf.Timeout == 0 {
+		conf.Timeout = time.Second * 5
+	}
+
+	return conf.FormatDSN(), nil
+}
```

Comment:
Why do you enforce a (undocumented) timeout? Either make this an config option or skip this completely as the user already has the option to specify a timeout using the DSN.